### PR TITLE
Feature/mob kinesis key

### PR DESCRIPTION
**Pull request by LBirk**
Here's the work for the collector kinesis partition key change. I also updated the branch to bring it up to date with the SP master branch. I'll call out my specific changes in the PR

### CODE REVIEW

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala [hunk 1]
```diff
@@ -54,43 +59,122 @@ class CollectorServiceActor(collectorConfig: CollectorConfig,
...
-        optionalCookie("sp") { reqCookie =>
+        cookieIfWanted(cookieName) { reqCookie =>
           optionalHeaderValueByName("User-Agent") { userAgent =>
             optionalHeaderValueByName("Referer") { refererURI =>
               headerValueByName("Raw-Request-URI") { rawRequest =>
-                hostName { host =>
-                  clientIP { ip =>
-                    requestInstance{ request =>
-                      entity(as[String]) { body =>
+                headerValueByName("SP-Cookie") { spCookie =>
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:87]: Review this

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/ResponseHandler.scala [hunk 2]
```diff
@@ -75,110 +98,155 @@ object ResponseHandler {
 // Receive requests and store data into an output sink.
 class ResponseHandler(config: CollectorConfig, sinks: CollectorSinks)(implicit context: ActorRefFactory) {
 
-  import context.dispatcher
-
   val Collector = s"${generated.Settings.shortName}-${generated.Settings.version}-" + config.sinkEnabled.toString.toLowerCase
 
   // When `/i` is requested, this is called and stores an event in the
   // Kinisis sink and returns an invisible pixel with a cookie.
   def cookie(queryParams: String, body: String, requestCookie: Option[HttpCookie],
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/ResponseHandler.scala:105]: Review all changes on this file below this comment

File: 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/PostSpec.scala [hunk 3]
```diff
@@ -62,8 +62,11 @@ collector {
   }
 
   cookie {
+    enabled = true
     expiration = 365 days
+    name = sp
     domain = "test-domain.com"
+    SP-Cookie = "666"
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/PostSpec.scala]: Review this

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala [hunk 4]
```diff
@@ -17,24 +17,29 @@ package collectors
 package scalastream
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:17]: Review this file

File: 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2014 Snowplow Analytics Ltd. All rights reserved.
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala:2]: Review this file

File: 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/PostSpec.scala [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/PostSpec.scala:1]: Review this file

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala [hunk 7]
```diff
@@ -54,43 +59,122 @@ class CollectorServiceActor(collectorConfig: CollectorConfig,
...
   def actorRefFactory = context
 
+  val cookieName = collectorConfig.cookieName
+
   // TODO: reduce code duplication here
   val collectorRoute = {
     post {
       path(Segment / Segment) { (path1, path2) =>
-        optionalCookie("sp") { reqCookie =>
+        cookieIfWanted(cookieName) { reqCookie =>
```
Comment (seanhall) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:83]: Is it safer to retain the 'optionalCookie' name but allow the arg?

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala [hunk 8]
```diff
@@ -54,43 +59,122 @@ class CollectorServiceActor(collectorConfig: CollectorConfig,
...
                             reqCookie,
                             userAgent,
                             host,
                             ip,
                             request,
                             refererURI,
-                            "/" + path1 + "/" + path2,
-                            false
+                            "/" + path,
+                            true,
```
Comment (seanhall) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:176]: What's the function of changing this boolean from false to true?

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala [hunk 9]
```diff
@@ -101,93 +185,69 @@ class CollectorService(
...
+      complete(responseHandler.notFound)
+  }
+
+  /**
+    * Directive to extract a cookie if a cookie name is specified and if such a cookie exists
+    *
+    * @param name Optionally configured cookie name
+    * @return Directive1[Option[HttpCookie]]
+    */
+  def cookieIfWanted(name: Option[String]): Directive1[Option[HttpCookie]] = name match {
```
Comment (seanhall) [inline 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:249]: Ah, it's its own new function, of course.

**Review by seanhall — COMMENTED** [review]
I think the only way I'll really know if this works is to put it into dev implementation, let's do that if we haven't already?

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala (hunk 8 above)
Comment (LBirk) [reply 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:176]: I didn't change this, this comes from the master mainline SP sync so I think we are stuck with it unless we want to diverge further from the mainline

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala (hunk 7 above)
Comment (LBirk) [reply 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:83]: I didn't change this, this comes from the master mainline SP sync so I think we are stuck with it unless we want to diverge further from the mainline

File: 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala (hunk 9 above)
Comment (LBirk) [reply 2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala:249]: Yeah, I'm going to get the collector running on a new dev box on Thursday to QA it further, if it works, we'll roll it out to all the dev boxes (blue-green style so there should be no downtime)

File: 3-enrich/scala-common-enrich/project/BuildSettings.scala [hunk 10]
```diff
@@ -21,9 +21,9 @@ object BuildSettings {
   // Basic settings for our app
   lazy val basicSettings = Seq[Setting[_]](
     organization          :=  "com.snowplowanalytics",
-    version               :=  "0.21.0",
+    version               :=  "1.24.0", //our Nordstrom version of common enrich
```
Comment (LBirk) [inline 3-enrich/scala-common-enrich/project/BuildSettings.scala:24]: I was unsure how to version this so I made a major version increment but kept the minor version so we could know which minor version we forked from. I'd love a better suggestion as to how this should be versioned

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterRegistry.scala [hunk 11]
```diff
@@ -76,6 +76,7 @@ object AdapterRegistry {
   def toRawEvents(payload: CollectorPayload)(implicit resolver: Resolver): ValidatedRawEvents = (payload.api.vendor, payload.api.version) match {
     case (Vendor.Snowplow,     "tp1") => SpTp1Adapter.toRawEvents(payload)
     case (Vendor.Snowplow,     "tp2") => SpTp2Adapter.toRawEvents(payload)
+    case (Vendor.Snowplow,     "mob") => SpTp2Adapter.toRawEvents(payload)
```
Comment (LBirk) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterRegistry.scala:79]: The new mobile endpoint!

File: 2-collectors/scala-stream-collector/project/Dependencies.scala [hunk 12]
```diff
@@ -33,11 +33,15 @@ object Dependencies {
     val awsSdk           = "1.6.10"
     val yodaTime         = "2.1"
     val yodaConvert      = "1.2"
+    val kafka            = "0.10.1.0"
     // Scala
+    val snowplowCommonEnrich = "0.22.0"
```
Comment (LBirk) [inline 2-collectors/scala-stream-collector/project/Dependencies.scala]: I was unsure how to version this so I made a major version increment but kept the minor version so we could know which minor version we forked from. I'd love a better suggestion as to how this should be versioned

**Review by atoadmin — APPROVED** [review]

**Pull request merged**